src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        FigureTypes type = FigureTypes.values()[random.nextInt(FigureTypes.values().length)];
+        switch (type) {
+            case SQUARE:
+                Square square = new Square();
+                square.setLeg(random.nextInt(1,100));
+                square.setColor(colorSupplier.getRandomColor());
+                return square;
```

Comment:
better to create a specific constructor for all figures to prevent creating figures without initialized fields